presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java

```diff
@@ -75,9 +88,136 @@ public boolean matches(long numberOfRows, Map<Integer, ColumnStatistics> statist
+        Optional<Map<C, Domain>> optionalStripeDomains = stripeDomain.getDomains();
+
+        // we need the effective predicate domains for bloom filter analysis
+        if (!optionalEffectivePredicateDomains.isPresent() || !optionalStripeDomains.isPresent()) {
+            // treat as failure: read
+            return true;
+        }
+
+        Map<C, Domain> effectivePredicateDomains = optionalEffectivePredicateDomains.get();
+        if (effectivePredicateDomains.isEmpty()) {
+            // no predicate, probably a where-less query, read
+            return true;
+        }
+        Map<C, Domain> stripDomains = optionalStripeDomains.get();
+
+        int numberOfBloomFiltersChecked = 0;
+        for (ColumnReference<C> columnReference : columnReferences) {
+            // is this part of a predicate?
+            if (!effectivePredicateDomains.containsKey(columnReference.getColumn())) {
+                continue;
+            }
+            if (!stripDomains.containsKey(columnReference.getColumn())) {
+                continue;
+            }
+
+            ColumnStatistics columnStatistics = statisticsByColumnIndex.get(columnReference.getOrdinal());
+            if (columnStatistics == null) {
+                // need column statistics, treat as failure: read
+                return true;
+            }
+
+            HiveBloomFilter bloomFilter = columnStatistics.getBloomFilter();
+            if (bloomFilter == null) {
+                // need bloom filters, treat as failure: read
+                return true;
+            }
+
+            Domain domain = effectivePredicateDomains.get(columnReference.getColumn());
+            if (domain.isAll()) {
+                // all values are okay, go to next one as we don't have to check the bf
+                continue;
+            }
+
+            // extract values from domain
+            Collection<Object> predicateValues = predicateValuesFromDomain(domain);
+            if (predicateValues == null || predicateValues.isEmpty()) {
+                // no values checked, treat as failure: read
+                return true;
+            }
+
+            // run values against the bloomFilter: none can match in order to skip a read
+            for (Object predicateValue : predicateValues) {
+                boolean matched = checkInBloomFilter(bloomFilter, predicateValue, domain.getType());
+                numberOfBloomFiltersChecked++;
+                if (matched) {
+                    // bloom filter is matched here return true so we select this stripe as it likely contains data which we need to read
+                    return true;
+                }
+            }
+        }
+
+        // did we read any bloom filter?
+        if (numberOfBloomFiltersChecked < 1) {
+            // no bloom filters read, seems like a query that was run only on partitioned columns
+            return true;
```

Comment:
No it shouldn't. If the bloom filters were unable to rule out reading this section we should read it: lack of bloom filters should be treated the same. One example was `select * from table where partition_col=xyz` which triggered an error in unit test of hive.